src/datatrove/pipeline/writers/disk_base.py

```diff
@@ -40,8 +40,13 @@ def __init__(
         self.compression = compression
         self.output_folder = get_datafolder(output_folder)
         output_filename = output_filename or self.default_output_filename
-        if self.compression == "gzip" and not output_filename.endswith(".gz"):
+        if output_filename.endswith(".parquet") and compression in ["snappy", "gzip", "brotli", "lz4", "zstd"]:
+            output_filename = output_filename.replace(".parquet", f".{compression}.parquet")
+            compression = None  # compression is already handled by ParquetWriter
+        elif self.compression == "gzip" and not output_filename.endswith(".gz"):
             output_filename += ".gz"
+        elif self.compression == "zstd" and not output_filename.endswith(".zstd"):
+            output_filename += ".zstd"
```

Comment:
Isn't .zst more commonly used?
```suggestion
        elif self.compression == "zstd" and not output_filename.endswith(".zst"):
            output_filename += ".zst"
```

Reply:
@guipenedo 
When I used PySpark, files were saved with the .zstd.parquet extension when using zstd compression. However, it seems that .zst is also a commonly used format. I’ve always used .zstd, but I’m not sure which one is more commonly used. 🤨

Reply:
I think official extension is ".zst" https://datatracker.ietf.org/doc/html/rfc8878#section-7.1

I will fix it